standards/async.md

```diff
@@ -0,0 +1,90 @@
+# Asynchronous
+
+## Overview
+
+There are number of situations where REST API endpoints require longer processing time to handle specific requests. Those type of particular operations have to run in the background outside of the request processing scope. These API endpoints should be designed to handle asynchronous communication in order to make the results of the request available to the consumer.
+
+There are couple ways to accomplish asynchronous communication in REST APIs and those are not mutually exclusive, including polling, websockets and webhooks.
+
+```note
+At this time only asynchronous communication through polling are provided in the API Standards.
+```
+
+## Polling (Resource without Status Monitor)
+
+Overall guidelines for asynchronous requests are:
+- API requests **SHOULD** be implemented as asynchronous if the 99th percentile response time is greater than 5s.
+- API requests **SHOULD NOT** decide or interpret whether a response will be asynchronous or synchronous dynamically at runtime. An endpoint should be either synchronous or asynchronous, but not both.
+- PATCH requests **MUST NOT** be used as asynchronous request initiation. If entity update is required and takes more than established time then it **MUST** be implemented as a POST request.
+
+All asynchronous endpoint requests start with initiation. Since long-running requests need to be processed on background, asynchronous requests should initiate those processes with necessary input parameters. Background process details should be returned.
+
+- Asynchronous requests **MUST** be a `POST` method, and should be represented as a resource, not an action-style endpoint.
```

Comment:
will we allow async on DELETE? if so, it doesn't work with the pattern where status monitor and resource are the same object.

Reply:
Why DELETE conflict with status monitor? I think background job can be cancelled, but not deleted.

Reply:
In the current pattern the status monitor and the resource are the same. So you'd have to make a GET request to the resource with a given id... and poll until you get a 404 on it?